manof.py

```diff
@@ -73,14 +73,34 @@ def _register_arguments(parser):
     # update
     subparsers.add_parser('update', help='Updates Manof')
 
+    # base sub parser
+    base_command_parent_parser = argparse.ArgumentParser(add_help=False)
+    base_command_parent_parser.add_argument('targets', nargs='+')
+    base_command_parent_parser.add_argument('-e',
+                                            '--exclude',
+                                            help='Exclude targets when running manof cmd (comma-delimited, no spaces)',
+                                            default='')
+
+    # TODO: Change default to 'docker.io'. Currently default is None for backwards compatibility
+    base_command_parent_parser.add_argument('-r',
+                                            '--repository',
+                                            help='The repository from which images shall be taken from or pushed to',
+                                            default=None)
+
+    # image based commands
+    image_base_parent_command = argparse.ArgumentParser(add_help=False)
+    image_base_parent_command.add_argument('-n', '--no-cache',
```

Comment:
should be associated with image provisioning only (no meaning for --no-cache for run / stop / rm)